flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/CodeGenerator.scala

```diff
@@ -148,16 +148,24 @@ class CodeGenerator(
       if (clazz == classOf[FlatMapFunction[_,_]]) {
         val inputTypeTerm = boxedTypeTermForTypeInfo(input1)
         (s"void flatMap(Object _in1, org.apache.flink.util.Collector $collectorTerm)",
-          s"$inputTypeTerm $input1Term = ($inputTypeTerm) _in1;")
+          List(s"$inputTypeTerm $input1Term = ($inputTypeTerm) _in1;"))
       }
 
       // MapFunction
       else if (clazz == classOf[MapFunction[_,_]]) {
         val inputTypeTerm = boxedTypeTermForTypeInfo(input1)
         ("Object map(Object _in1)",
-          s"$inputTypeTerm $input1Term = ($inputTypeTerm) _in1;")
+          List(s"$inputTypeTerm $input1Term = ($inputTypeTerm) _in1;"))
       }
 
+      // FlatJoinFunction
+      else if (clazz == classOf[FlatJoinFunction[_,_,_]]) {
+        val inputTypeTerm1 = boxedTypeTermForTypeInfo(input1)
+        val inputTypeTerm2 = boxedTypeTermForTypeInfo(input2.get) //TODO check this
```

Comment:
Why don't you call `getOrElse` here and throw an CodeGenException if input2 is not present? Then we can get rid of the TODO.